tgui/packages/tgui/interfaces/Autolathe.js

```diff
@@ -0,0 +1,272 @@
+        ))}
+      </Flex>
+      <CurrentlyMaking />
+    </Section>
+  );
+};
+
+const CurrentlyMaking = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { currently_making } = data;
+
+  return currently_making ? (
+    <Fragment>
+      <Box height="5px" />
+      <Button
+        fluid
+        textAlign="center"
+        content={
+          'Currently making: ' +
+          capitalize(currently_making.name) +
+          (currently_making.multiplier > 1
+            ? ' (x' + currently_making.multiplier + ')'
+            : '')
+        }
+      />
+    </Fragment>
+  ) : null;
+};
+
+const QueueList = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    materials,
+    capacity,
+    queued,
+    printables,
+    selectable_categories,
+    queuemax,
+  } = data;
+
+  return (
+    <Section title="Queue">
+      <Flex direction="column">
+        {queued.map((item) => {
+          return (
+            <Flex.Item key={item}>
+              <Flex direction="row">
+                <Flex.Item>
+                  <Button
+                    content={
+                      item.index +
+                      ': ' +
+                      capitalize(item.name) +
+                      (item.multiplier > 1 ? ' (x' + item.multiplier + ')' : '')
+                    }
+                  />
+                  <Box width="5px" />
+                </Flex.Item>
+                <Flex.Item>
+                  <Button
+                    icon="xmark"
+                    tooltip="Remove"
+                    onClick={() =>
+                      act('cancel', {
+                        index: item.index,
+                        name: item.name,
+                        multiplier: item.multiplier,
+                      })
+                    }
+                  />
+                </Flex.Item>
+              </Flex>
+            </Flex.Item>
+          );
+        })}
+      </Flex>
+    </Section>
+  );
+};
+
+// the below all has to be in one section due to the categories and search params
+const PrintablesSection = (props, context) => {
+  const { act, data } = useBackend(context);
+
+  const logger = createLogger('autolathe');
+
+  const {
+    materials,
+    capacity,
+    queued,
+    printables,
+    selectable_categories,
+    queuemax,
+  } = data;
+
+  const [currentSearch, setSearch] = useLocalState(
+    context,
+    'current_search',
+    ''
+  );
+
+  const categories = [];
+  for (let i = 0; i < printables.length; i++) {
+    let data = printables[i];
+    if (categories.includes(data.recipe_category)) continue;
+
+    categories.push(data.recipe_category);
+  }
+
+  const [currentCategory, setCategory] = useLocalState(
+    context,
+    'current_category',
+    'All'
+  );
+
+  const filteredPrintables = printables.filter(
+    (val) =>
+      (val.recipe_category === currentCategory || currentCategory === 'All') &&
+      val.name.toLowerCase().match(currentSearch)
```

Comment:
I believe the currently accepted way to do a case insensitive search (besides regex) is localeCompare https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare - it also has good non-English language support

Reply:
couldn't seem to get this working properly